internal/handler/favourites.go

```diff
@@ -0,0 +1,296 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"svenvermeulen/platform-go-challenge/internal/auth"
+	"svenvermeulen/platform-go-challenge/internal/repository/audience"
+	"svenvermeulen/platform-go-challenge/internal/repository/chart"
+	"svenvermeulen/platform-go-challenge/internal/repository/favourite"
+	"svenvermeulen/platform-go-challenge/internal/repository/insight"
+	"svenvermeulen/platform-go-challenge/pkg/model"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+type FavouritesHandler struct {
+	favouriteRepository *favourite.Repository
```

Comment:
Using interfaces there would allow us to swap out repository implementations without changing the handlers.
Examples would be separate implementations for different data stores, or injecting a mock implementation for testing purposes.